packages/k8s/src/k8s/utils.ts

```diff
@@ -111,11 +111,11 @@ export function writeEntryPointScript(
   if (environmentVariables && Object.entries(environmentVariables).length) {
     const envBuffer: string[] = []
     for (const [key, value] of Object.entries(environmentVariables)) {
+      if (key.includes(`=`) || key.includes(`'`) || key.includes(`"`)) {
+        throw new Error(`environment key ${key} is invalid`)
```

Comment:
```suggestion
        throw new Error(`environment key ${key} is invalid - the key must not contain =, ' or "`)
```

NIT: If this exception can be hit on genuine user error, we could be more explicit on our rules - if you don't want to expose the implementation it's fine too.